pkg/client/unversioned/urlbackoff.go

```diff
@@ -95,3 +99,7 @@ func (b *URLBackoff) UpdateBackoff(actualUrl *url.URL, err error, responseCode i
 func (b *URLBackoff) CalculateBackoff(actualUrl *url.URL) time.Duration {
 	return b.Backoff.Get(b.baseUrlKey(actualUrl))
 }
+
+func (b *URLBackoff) Sleep(d time.Duration) {
```

Comment:
Sleep is implemented inside urlbackoff to decouple the request from any notion of a depedency on the clock interface.  This is important because I think we may want to swap out the clock later for an external clock impl (i.e. https://github.com/jonboulle/clockwork)